Cargo.toml

```diff
@@ -595,6 +595,7 @@ wasmtime-wasi = "29"
 which = "6.0.0"
 wit-component = "0.221"
 zed_llm_client = "0.4"
+zip = "2.5.0"
```

Comment:
Why are we using the `zip` crate?

This diverges from how we deal with archives everywhere else in Zed.

Reply:
I am using it because the official releases of the lsp are in zip format instead of gzip as before.

Reply:
I checked now and see that async_zip is used in other places like crates/norde_runtime/src/archive.ts. I will use that instead

Reply:
We have existing ways of extracting zip files in Zed. We should be following our existing patterns.

Reply:
Will double check next time. Should I also revert the lock file changes?

Reply:
I ended up refactoring any remaining bits, also merged your change to get the lsp from npm. Please let me know if any further modifications are

Reply:
Is there any update on this? Noticed the latest release included the subtle/ eager mode so I believe this feature should complement nice!